astroquery/hsa/core.py

```diff
@@ -0,0 +1,356 @@
+            example: 1342195355
+        retrieval_type : string, optional, default 'OBSERVATION'
+            The type of product that we want to retrieve
+            values: OBSERVATION, PRODUCT, POSTCARD, POSTCARDFITS, REQUESTFILE_XML, STANDALONE, UPDP, HPDP
+        instrument_name : string, optional, default 'PACS'
+            values: PACS, SPIRE, HIFI
+            The instrument name, by default 'PACS' if the retrieval_type is 'OBSERVATION'
+        filename : string, optional, default None
+            If the filename is not set it will use the observation_id as filename
+            file name to be used to store the file
+        verbose : bool, optional, default False
+            flag to display information about the process
+        observation_oid : string, optional
+            Observation internal identifies. This is the database identifier
+        instrument_oid : string, optional
+            The database identifies of the instrument
+            values: 1, 2, 3
+        product_level : string, optional
+            level to download
+            values: ALL, AUXILIARY, CALIBRATION, LEVEL0, LEVEL0_5, LEVEL1, LEVEL2, LEVEL2_5, LEVEL3, ALL-LEVEL3
+
+        Returns
+        -------
+        File name of downloaded data
+        """
+        if filename is not None:
+            filename = os.path.splitext(filename)[0]
+
+        if retrieval_type is None:
+            retrieval_type = "OBSERVATION"
+
+        params = {'retrieval_type': retrieval_type}
+        if observation_id is not None:
+            params['observation_id'] = observation_id
+
+        if retrieval_type == "OBSERVATION" and instrument_name is None:
+            instrument_name = "PACS"
+
+        if instrument_name is not None:
+            params['instrument_name'] = instrument_name
+
+        link = self.data_url + "".join("&{0}={1}".format(key, val)
+                                       for key, val in params.items())
+
+        link += "".join("&{0}={1}".format(key, val)
+                        for key, val in kwargs.items())
+
+        if verbose:
+            log.info(link)
+
+        response = self._request('HEAD', link, save=False, cache=cache)
+        response.raise_for_status()
+
+        if 'Content-Type' in response.headers and 'text' not in response.headers['Content-Type']:
+            _, params = cgi.parse_header(response.headers['Content-Disposition'])
+        else:
+            error = "Data protected by propietary rights. Please check your credentials"
+            raise LoginError(error)
+
+        if filename is None:
+            if observation_id is not None:
+                filename = observation_id
+            else:
+                error = "Please set a filename for the output"
+                raise InputWarning(error)
+
+        r_filename = params["filename"]
+        suffixes = Path(r_filename).suffixes
+
+        if len(suffixes) > 1 and suffixes[len(suffixes) - 1] == ".jpg":
```

Comment:
```suggestion
        if len(suffixes) > 1 and suffixes[-1] == ".jpg":
```
`len(x) - 1` is equivalent to `-1`